File: backend/app/schemas/system.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field, field_validator


class SystemSettingsItem(BaseModel):
    site_name: str
    site_subtitle: str
    logo_url: str | None = None
    brand_accent: str
    enable_ai_scoring: bool
    enable_export_center: bool
    enable_audit_enhanced: bool
    admin_menu_template_lock: bool
    faq_task_timeout_minutes: int
    updated_at: datetime


class PublicSystemSettingsItem(BaseModel):
    site_name: str
    site_subtitle: str
    logo_url: str | None = None
    brand_accent: str
    default_tenant_code: str | None = None
    show_tenant_input: bool = False


class SystemSettingsUpdateRequest(BaseModel):
    site_name: str = Field(min_length=2, max_length=120)
    site_subtitle: str = Field(min_length=2, max_length=120)
    logo_url: str | None = Field(default=None, max_length=500)
    brand_accent: str = Field(min_length=4, max_length=20)
    enable_ai_scoring: bool
    enable_export_center: bool
    enable_audit_enhanced: bool
    admin_menu_template_lock: bool
    faq_task_timeout_minutes: int = Field(default=15, ge=5, le=30)

    @field_validator("faq_task_timeout_minutes")
    @classmethod
    def validate_faq_task_timeout_minutes(cls, v: int) -> int:
        if v not in {5, 15, 30}:
            raise ValueError("FAQ 任务超时阈值仅支持 5/15/30 分钟")
        return v


class ApiResponse(BaseModel):
    code: int
    message: str
    data: SystemSettingsItem | PublicSystemSettingsItem | None = None

File: frontend/src/api/system.ts
import request from './request'
import type { AxiosRequestConfig } from 'axios'

interface ApiResponse<T> {
  code: number
  message: string
  data: T
}

export interface PublicSystemSettings {
  site_name: string
  site_subtitle: string
  logo_url?: string | null
  brand_accent: string
  default_tenant_code?: string | null
  show_tenant_input?: boolean
}

export interface SystemSettings extends PublicSystemSettings {
  enable_ai_scoring: boolean
  enable_export_center: boolean
  enable_audit_enhanced: boolean
  admin_menu_template_lock: boolean
  faq_task_timeout_minutes: 5 | 15 | 30
  updated_at: string
}

export interface SystemSettingsUpdatePayload {
  site_name: string
  site_subtitle: string
  logo_url?: string | null
  brand_accent: string
  enable_ai_scoring: boolean
  enable_export_center: boolean
  enable_audit_enhanced: boolean
  admin_menu_template_lock: boolean
  faq_task_timeout_minutes: 5 | 15 | 30
}

export const getPublicSystemSettings = async () => {
  const res = await request.get<ApiResponse<PublicSystemSettings>>('/system/settings/public')
  return res.data.data
}

export const getSystemSettings = async (params?: { tenant_id?: number }) => {
  const res = await request.get<ApiResponse<SystemSettings>>('/system/settings', { params })
  return res.data.data
}

export const updateSystemSettings = async (
  payload: SystemSettingsUpdatePayload,
  params?: { tenant_id?: number },
) => {
  const res = await request.put<ApiResponse<SystemSettings>>('/system/settings', payload, { params })
  return res.data.data
}

const exportBlob = async (url: string, params?: Record<string, string | number | undefined>) => {
  const config: AxiosRequestConfig = {
    params,
    responseType: 'blob',
  }
  const res = await request.get(url, config)
  return res.data as Blob
}

export const exportUsersCsv = async (params?: {
  hospital_id?: number
  department_id?: number
  role?: string
  start_date?: string
  end_date?: string
}) => exportBlob('/system/exports/users', params)

export const exportPracticesCsv = async (params?: {
  hospital_id?: number
  department_id?: number
  status?: string
  start_date?: string
  end_date?: string
}) => exportBlob('/system/exports/practices', params)

export const exportQuizzesCsv = async (params?: {
  hospital_id?: number
  department_id?: number
  scope?: string
  start_date?: string
  end_date?: string
}) => exportBlob('/system/exports/quizzes', params)
